src/rgw/rgw_lua_request.cc

```diff
@@ -315,19 +304,21 @@ struct BucketMetaTable : public EmptyMetaTable {
     } else if (strcasecmp(index, "MTime") == 0) {
       pushtime(L, bucket->get_modification_time());
     } else if (strcasecmp(index, "Quota") == 0) {
-      create_metatable<QuotaMetaTable>(L, false, &(bucket->get_info().quota));
+      create_metatable<QuotaMetaTable>(L, name, index, false, &(bucket->get_info().quota));
     } else if (strcasecmp(index, "PlacementRule") == 0) {
-      create_metatable<PlacementRuleMetaTable>(L, false, &(bucket->get_info().placement_rule));
+      create_metatable<PlacementRuleMetaTable>(L, name, index, false, &(bucket->get_info().placement_rule));
     } else if (strcasecmp(index, "User") == 0) {
-      create_metatable<UserMetaTable>(L, false, &(bucket->get_info().owner));
+      create_metatable<UserMetaTable>(L, name, index, false, 
+          const_cast<rgw_user*>(&bucket->get_owner()->get_id()));
```

Comment:
what was broken here?

`const_cast` is a common source of undefined behavior, and can almost always be avoided by using const correctly. does a non-const user here mean that lua can modify a bucket's owner? i don't think we want to allow that

p.s. why Bucket.User instead of Bucket.Owner?

Reply:
> a change in zipper caused `bucket->get_info().owner` to return an empty string.

do you know where that changed? can we please fix that too? cc @dang